src/main/java/com/files/management/service/LocationService.java

```diff
@@ -2,27 +2,47 @@
 
 import com.files.management.entity.Location;
 import com.files.management.exception.DuplicateLocationException;
+import com.files.management.exception.LocationNotFoundException;
 import com.files.management.mapper.LocationMapper;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 import org.springframework.validation.annotation.Validated;
 
 @Validated
 @Service
 public class LocationService {
-    private final LocationMapper locationMapper;
 
-    public LocationService(LocationMapper locationMapper){
-        this.locationMapper = locationMapper;
+  private final LocationMapper locationMapper;
+
+  public LocationService(LocationMapper locationMapper) {
+    this.locationMapper = locationMapper;
+  }
+
+  public Location insert(String locationName, String shelfNumber) {
+    // 複合ユニークバリデーション
+    if (locationMapper.isMaterialUnique(locationName, shelfNumber)) {
+      throw new DuplicateLocationException("Location with location:" + locationName
+          + " and shelfNumber:" + shelfNumber + " already exists");
     }
```

Comment:
isMaterialUniqueというメソッド名だと、ユニークである場合（ユニークなことがtrueだったら）、重複となり例外をスローする、という矛盾した命名になってしまっています。

以下のようにすることで、カウントが0以上だった場合に例外をスローするという記述になり実態に即した名前になっているかなと思います！（ちょっと命名が長いので工夫しても良いかもしれないですが、思いつきませんでした）
```suggestion
    if (locationMapper.getCountByLocationNameAndShelfNumber(locationName, shelfNumber) > 0) {
      throw new DuplicateLocationException("Location with location:" + locationName
          + " and shelfNumber:" + shelfNumber + " already exists");
    }
```

Reply:
たしかに！ユニークでないことを検証しているのでおかしいですね。
そしてここにも参考にしコードの情報が残っていました。。修正いたします。

Reply:
こちらも同様です
a96d39571a1f3b448224f907d68ecb3848667a12